### PR TITLE
feature/#115 add a check box for household items in the visit form

### PR DESCRIPTION
**Pull request by lmeng25**
Added a household item checkbox on the shopping sheet
image

Added household item labels on visit form, and visit details
image
image

Added behavior in visit form: uncheck the household item checkbox will reset household item notes. 

issue #115

### CODE REVIEW

File: components/Visit/VisitInfoForm.tsx [hunk 1]
```diff
@@ -210,17 +234,33 @@ export default function VisitInfoForm({
...
-            />
+            <div hidden={!visitData.householdItem}>
+                <h2>Household items</h2>
+                <FormItem
+                    type="textarea"
+                    id="household"
+                    rows={3}
+                    value={visitData.household || ''}
+                    onChange={handleChange('household')}
+                    // hidden={!visitData.householdItem}
```
Comment (ducmvo) [inline components/Visit/VisitInfoForm.tsx:261]: remove comment here

File: components/Visit/VisitInfoForm.tsx [hunk 2]
```diff
@@ -88,9 +90,31 @@ export default function VisitInfoForm({
...
+        }
+
+        if (onChange) {
+            const updatedData = {
+                ...visitData,
+                [key]: value,
+                ...(key === 'householdItem' && !value && { household: '' }),
+            };
+            onChange(transformData(updatedData));
+        }
```
[the remark this answers is not in the document]
Comment (lmeng25) [reply components/Visit/VisitInfoForm.tsx:117]: Thanks for pointing it out! User may need to uncheck the household item checkbox manually every time if we default the flag to true. Instead, we can update the initialVisitData.household depending on whether the household note is empty.

File: components/Visit/VisitInfoForm.tsx [hunk 3]
```diff
@@ -88,9 +91,31 @@ export default function VisitInfoForm({
...
```
Comment (ducmvo) [inline components/Visit/VisitInfoForm.tsx:118]: could we move this block to into  `transformData` function?
We can have a simple one line "check and set" value similar to line 79:
```ts
kidsQ: visitData.clothingBoy || visitData.clothingGirl ? toInt(visitData.kidsQ) : 0
```

can we do something like:
```ts
household: visitData.houseHoldItem ? visitData.household : ''
```

Comment (lmeng25) [reply components/Visit/VisitInfoForm.tsx:118]: Yes, that'd be more clean and clear! 
I also added initialVisitData?.householdItem for a condition in useState for the case that the checkbox is checked but note is empty.

**Review by ducmvo — APPROVED** [review]
Good work!

**Pull request merged**